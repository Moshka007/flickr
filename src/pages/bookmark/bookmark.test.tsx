import React from "react";
import { render, screen } from "@testing-library/react";
import Bookmark from "./bookmark";


const bookmarkArray: {id: number, url_s: string, tags: string[]}[] = [
    {
        id: 1,
        url_s: '#',
        tags: ['1', '2', '3']
    },
    {
        id: 1,
        url_s: '#',
        tags: ['1', '2', '3']
    }],

    setB = jest.fn();


describe('Testing <Bookmark/>', () => {
    it('Bookmark have rendered correctly', () => {
        const pages = render(<Bookmark bookMark={true} bookmarkArray={bookmarkArray} setBookMarkArray={setB} />);
        expect(pages).toMatchSnapshot();
    });

    it('PhotoListItem render correctly', () => {
        render(<Bookmark bookMark={true} bookmarkArray={bookmarkArray} setBookMarkArray={setB} />);
        expect(screen.getAllByRole('img').length).toEqual(2);
    })
})
