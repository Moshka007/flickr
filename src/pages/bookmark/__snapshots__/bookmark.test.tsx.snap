// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Bookmark/> Bookmark have rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="photo-list row"
        style="overflow-y: scroll; overflow-x: hidden; height: calc(100vh - 8em); margin-top: 1em;"
      >
        <div
          class="pt-5 col-md-4"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="#"
              style="padding: 1.5em 1em 0px 1em;"
            />
            <div
              class="card-body"
            >
              <button
                class="btn btn-light"
                style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
                type="button"
              >
                Remove it
              </button>
            </div>
          </div>
        </div>
        <div
          class="pt-5 col-md-4"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="#"
              style="padding: 1.5em 1em 0px 1em;"
            />
            <div
              class="card-body"
            >
              <button
                class="btn btn-light"
                style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
                type="button"
              >
                Remove it
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="photo-list row"
      style="overflow-y: scroll; overflow-x: hidden; height: calc(100vh - 8em); margin-top: 1em;"
    >
      <div
        class="pt-5 col-md-4"
      >
        <div
          class="card"
        >
          <img
            class="card-img-top"
            src="#"
            style="padding: 1.5em 1em 0px 1em;"
          />
          <div
            class="card-body"
          >
            <button
              class="btn btn-light"
              style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
              type="button"
            >
              Remove it
            </button>
          </div>
        </div>
      </div>
      <div
        class="pt-5 col-md-4"
      >
        <div
          class="card"
        >
          <img
            class="card-img-top"
            src="#"
            style="padding: 1.5em 1em 0px 1em;"
          />
          <div
            class="card-body"
          >
            <button
              class="btn btn-light"
              style="cursor: pointer; border: 1px solid black; opacity: 0.7;"
              type="button"
            >
              Remove it
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
